diff --git a/kstars/dialogs/finddialog.cpp b/kstars/dialogs/finddialog.cpp
index 77df85c49..749e33e7a 100644
--- a/kstars/dialogs/finddialog.cpp
+++ b/kstars/dialogs/finddialog.cpp
@@ -25,6 +25,7 @@
 #include <QTimer>
 #include <QComboBox>
 #include <QLineEdit>
+#include <QPointer>
 
 FindDialog *FindDialog::m_Instance = nullptr;
 
diff --git a/kstars/dialogs/fovdialog.cpp b/kstars/dialogs/fovdialog.cpp
index d45727967..d8a0b6a41 100644
--- a/kstars/dialogs/fovdialog.cpp
+++ b/kstars/dialogs/fovdialog.cpp
@@ -18,6 +18,7 @@
 #include <QLineEdit>
 #include <QDBusReply>
 #include <QDBusInterface>
+#include <QPointer>
 
 #include <KActionCollection>
 #include <KLocalizedString>
